rs/types/management_canister_types/src/lib.rs

```diff
@@ -989,6 +989,8 @@ pub struct CanisterStatusResultV2 {
     controller: candid::Principal,
     settings: DefiniteCanisterSettingsArgs,
     memory_size: candid::Nat,
+    execution_memory_size: ExecutionMemorySize,
+    system_memory_size: SystemMemorySize,
```

Comment:
This makes sense to break it up this way, but I wonder if it's a bit too low level, in terms of exposing what is execution memory vs. system memory. It might be simpler to just have one record, MemoryMetrics or MemoryFootprint, that contains all of these fields:

Open for feedback here, but I might also rename `wasm_memory_size` to `heap_size`, given that it's already in the `MemoryProfile` for consistency with how it's communicated in all developer facing documentation.

```
pub struct MemoryMetrics {
  heap_size: candid::Nat,
  // ...all other memory components (both system and execution)
  snapshot_size: candid::Nat,
}
```